tests/components/light/test_zwave.py

```diff
@@ -105,6 +105,26 @@ def test_dimmer_turn_on(mock_openzwave):
         assert entity_id == device.entity_id
 
 
+def test_dimmer_min_brightness(mock_openzwave):
+    """Test turning on a dimmable Z-Wave light to its minimum brightness."""
+    node = MockNode()
+    value = MockValue(data=0, node=node)
+    values = MockLightValues(primary=value)
+    device = zwave.get_device(node=node, values=values, node_config={})
+
+    assert not device.is_on
+
+    device.turn_on(**{ATTR_BRIGHTNESS: 1})
+
+    assert device.is_on
+    assert device.brightness == 1
+
+    device.turn_on(**{ATTR_BRIGHTNESS: 0})
+
+    assert device.is_on
+    assert device.brightness == 0
```

Comment:
@armills I thought about the second part of this test here that I added some more, and I'd like your opinion as to whether or not the result makes sense. Right now, calling `turn on` with a brightness val of 0, will result in device.is_on returning true... Is that the intended behavior? If not, I can add an else block to the turn_on method that sets the state to off in the case of 0 being passed in.

Reply:
Good catch. That really should be cleaned up.

If you don't mind we could also refactor it to make it a little simpler. `self._state` can just be a boolean. `STATE_ON` can be replaced with `True`, `STATE_OFF` can be replaced with `False`, and we can return it directly here: https://github.com/home-assistant/home-assistant/blob/a40a6e31ada250d48241698ed133a847f45508a0/homeassistant/components/light/zwave.py#L154 . Then in the turn_on method you can just assign `self._state = brightness > 0`.